config/detekt/detekt.yml

```diff
@@ -63,4 +63,12 @@ style:
   SpacingBetweenPackageAndImports:
     active: true
   UnusedImports:
-    active: true
\ No newline at end of file
+    active: true
+
+  # Avoid business logic in payload/config classes
+  DataClassShouldBeImmutable:
+    active: true
+    includes: [ '**/io/embrace/android/embracesdk/payload/**' ]
```

Comment:
Do we want to remove the restrictions or are there too many data classes we are mutating now?

Reply:
I decided to confine this to payload/config as that's where most of the data classes should be where we care about business logic not being present